src/spikeinterface/sortingcomponents/tests/test_benchmark_matching.py

```diff
@@ -0,0 +1,179 @@
+                sorting = sc.NumpySorting.from_times_labels(
+                    spike_time_inds, unit_ids, benchmark.recording.sampling_frequency
+                )
+                spike_time_inds = rng.choice(benchmark.recording.get_num_frames(), len_spike_train, replace=False)
+                unit_ids = rng.choice(benchmark.gt_sorting.get_unit_ids(), len_spike_train, replace=True)
+                sort_index = np.argsort(spike_time_inds)
+                spike_time_inds = spike_time_inds[sort_index]
+                unit_ids = unit_ids[sort_index]
+                gt_sorting = sc.NumpySorting.from_times_labels(
+                    spike_time_inds, unit_ids, benchmark.recording.sampling_frequency
+                )
+                sortings.append(sorting)
+                gt_sortings.append(gt_sorting)
+                parameter_values.append(spike_num)
+                parameter_names.append(parameter_name)
+                iter_nums.append(replicate)
+                methods.append(method)
+    matching_df = pd.DataFrame(
+        {
+            "sorting": sortings,
+            "gt_sorting": gt_sortings,
+            "parameter_value": parameter_values,
+            "parameter_name": parameter_names,
+            "iter_num": iter_nums,
+            "method": methods,
+        }
+    )
+    comparison_from_df = matching_df.copy()
+    comparison_from_self = matching_df.copy()
+    comparison_collision = matching_df.copy()
+
+    # Act
+    benchmark.compare_all_sortings(comparison_from_df, ground_truth="from_df")
+    benchmark.compare_all_sortings(comparison_from_self, ground_truth="from_self")
+    benchmark.compare_all_sortings(comparison_collision, collision=True)
+
+    # Assert
+    for comparison in [comparison_from_df, comparison_from_self, comparison_collision]:
+        assert comparison.shape[0] == len(num_spikes) * num_replicates * len(methods_kwargs)
+        assert comparison.shape[1] == 7
+        for comp, sorting in zip(comparison["comparison"], comparison["sorting"]):
+            comp.sorting2 == sorting
+    for comp, gt_sorting in zip(comparison_from_df["comparison"], comparison["gt_sorting"]):
+        comp.sorting1 == gt_sorting
+    for comp in comparison_from_self["comparison"]:
+        comp.sorting1 == benchmark.gt_sorting
+
+
+def test_compare_all_sortings_invalid_inputs(benchmark_and_kwargs):
+    benchmark, methods_kwargs = benchmark_and_kwargs
+    with pytest.raises(ValueError):
+        benchmark.compare_all_sortings(pd.DataFrame(), ground_truth="invalid")
+
+
+if __name__ == "__main__":
+    test_run_matching_vary_parameter(benchmark_and_kwargs)
+    test_run_matching_vary_parameter_invalid_inputs(benchmark_and_kwargs)
+    test_run_matching_misclassed_invalid_inputs(benchmark_and_kwargs)
+    test_run_matching_missing_units_invalid_inputs(benchmark_and_kwargs)
+    test_compare_all_sortings(benchmark_and_kwargs)
+    test_compare_all_sortings_invalid_inputs(benchmark_and_kwargs)
```

Comment:
I am OK to keep this file but I would skip this on github for now.
This will avoid deep and complicated errors when refactoring sorting component, I do not think we want to maintain also benchmark at the same time.

Reply:
Isn't it easier to maintain if you know when you introduce errors?

Reply:
OK.
Lets keep this and I will make a decission on keep or not this tests running later.